### PR TITLE
Implement a page which lists all restaurants (frontend + backend)

### PR DESCRIPTION
**Pull request by limli**
Screenshot 2020-07-24 at 10 22 53 AM

### CODE REVIEW

File: src/main/java/com/google/step/datamanager/RestaurantManagerDatastore.java [hunk 1]
```diff
@@ -156,4 +156,15 @@ public void deleteAllRestaurants() {
             .collect(Collectors.toList());
     return restaurants;
   }
+
+  @Override
+  public List<Restaurant> getAllRestaurants() {
+    List<Restaurant> restaurants = new ArrayList<>();
+    Query query = new Query("Restaurant");
+    PreparedQuery pq = datastore.prepare(query);
+    for (Entity dealEntity : pq.asIterable()) {
```
Comment (jyx11011) [inline src/main/java/com/google/step/datamanager/RestaurantManagerDatastore.java]: restaurantEntity

File: src/main/webapp/js/all-restaurants.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  restaurantBody.appendChild(restaurantName);
+  restaurantBody.appendChild(restaurantLink);
+
+  restaurantCard.appendChild(restaurantImage);
+  restaurantCard.appendChild(restaurantBody);
+  return restaurantCard;
+}
+
+/**
+   * Calls backend on data on restaurants
```
Comment (jyx11011) [inline src/main/webapp/js/all-restaurants.js]: Delete extra spaces

**Review by jyx11011 — APPROVED** [review]

File: src/main/java/com/google/step/servlets/RestaurantPostListServlet.java [hunk 3]
```diff
@@ -40,10 +37,16 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
 
     Restaurant restaurant = manager.createRestaurantWithBlobKey(name, photoBlobkey);
-    List<Deal> deals = new ArrayList<>();
-    List<String> placeIds = Arrays.asList(places.split(","));
-    response.getWriter().println(JsonFormatter.getRestaurantJson(restaurant, deals, placeIds));
```
Comment (SuyashLakhotia) [inline src/main/java/com/google/step/servlets/RestaurantPostListServlet.java:45]: Just checking - this is never used on the frontend?

**Review by SuyashLakhotia — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/step/servlets/RestaurantPostListServlet.java (hunk 3 above)
Comment (limli) [reply src/main/java/com/google/step/servlets/RestaurantPostListServlet.java:45]: yes, because it redirects to the restaurant page, so the json isn't used

**Review by SuyashLakhotia — APPROVED** [review]

**Pull request merged**